### PR TITLE
Add support for files in submodules when using BCommits/Commits

### PR DESCRIPTION
**Pull request by matawed**
If file in buffer is located in submodule then bcommit doesn't show the history of this file.

fixes: https://github.com/junegunn/fzf.vim/issues/918

### CODE REVIEW

File: autoload/fzf/vim.vim [hunk 1]
```diff
@@ -1104,20 +1104,19 @@ function! s:commits(buffer_local, args)
   if empty(s:git_root)
     return s:warn('Not in git repository')
   endif
-
-  let source = 'git log '.get(g:, 'fzf_commits_log_options', '--color=always '.fzf#shellescape('--format=%C(auto)%h%d %s %C(green)%cr'))
+  let path = expand('%:p:h')
```
Comment (junegunn) [inline autoload/fzf/vim.vim]: We should wrap it with `fzf#shellescape` for it to work on different platforms.

File: autoload/fzf/vim.vim [hunk 2]
```diff
@@ -1104,20 +1104,19 @@ function! s:commits(buffer_local, args)
...
     call system('git show '.fzf#shellescape(current).' 2> '.(s:is_win ? 'nul' : '/dev/null'))
     let managed = !v:shell_error
   endif
-
   if a:buffer_local
     if !managed
       return s:warn('The current buffer is not in the working tree')
     endif
-    let source .= ' --follow '.fzf#shellescape(current)
+    let source .= ' --follow '.expand('%:t')
```
Comment (junegunn) [inline autoload/fzf/vim.vim]: `fzf#shellescape` is still needed here too.

Comment (matawed) [reply autoload/fzf/vim.vim]: Reverted change as it was not needed.

File: autoload/fzf/vim.vim (hunk 1 above)
Comment (matawed) [reply autoload/fzf/vim.vim]: done.

**Comment by junegunn** [comment]
Thanks, but I had to revert the change as `BCommits` stopped working for me. Have you tested it with a file on a repository without submodules?

**Comment by matawed** [comment]
It seems my tests weren't deep enough. For files with some (one?) change(s) it works. See e.g. 

image

vs

image

Will check.

**Pull request merged**